### PR TITLE
adding additional error codes

### PR DESCRIPTION
**Pull request by KarishmaGhiya**
Adding additional error codes check for link verification and checking inner exception codes for socket exceptions
Defaults to 400, 401, 404, SocketError.HostNotFound = 11001, SocketError.NoData = 11004

### CODE REVIEW

File: eng/common/scripts/Verify-Links.ps1 [hunk 1]
```diff
@@ -145,6 +145,10 @@ function CheckLink ([System.Uri]$linkUri)
     catch {
       $statusCode = $_.Exception.Response.StatusCode.value__
 
+      if(!$statusCode){
```
Comment (weshaggard) [inline eng/common/scripts/Verify-Links.ps1:148]: nit: add extra space before `{`

Comment (weshaggard) [reply eng/common/scripts/Verify-Links.ps1:148]: Can you also add a comment here that the ErrorCode is common when there is a SocketExeception and so we are try to pull the ErrorCode from the inner SocketException if it exists.

Comment (KarishmaGhiya) [reply eng/common/scripts/Verify-Links.ps1:148]:
```suggestion
      if(!$statusCode) {
```

File: eng/common/scripts/Verify-Links.ps1 [hunk 2]
```diff
@@ -145,6 +145,10 @@ function CheckLink ([System.Uri]$linkUri)
     catch {
       $statusCode = $_.Exception.Response.StatusCode.value__
 
+      if(!$statusCode) {
+        $statusCode = $_.Exception.InnerException.ErrorCode
```
Comment (weshaggard) [inline eng/common/scripts/Verify-Links.ps1:149]: Can you also add a comment here that the ErrorCode is common when there is a SocketExeception and so we are try to pull the ErrorCode from the inner SocketException if it exists.

**Comment by weshaggard** [comment]
FYI @sima-zhu

File: eng/common/scripts/Verify-Links.ps1 (hunk 2 above)
Comment (KarishmaGhiya) [reply eng/common/scripts/Verify-Links.ps1:149]:
```suggestion
        #Pulling the error code from the inner socket exception
        $statusCode = $_.Exception.InnerException.ErrorCode
```

**Pull request merged**